File: lua/chronopulser_cli.lua
io.write("Mode (encode/decode): ")
local mode = io.read():lower()
io.write("Input: ")
local input = io.read()

if mode=="encode" then
    dofile("chronopulse_encoder.lua")
elseif mode=="decode" then
    dofile("chronopulse_decoder.lua")
else
    print("Invalid mode")
end

File: csharp/ChronoPulserCLI.cs
using System;

class ChronoPulseCLI
{
    static void Main()
    {
        Console.Write("Choose mode (encode/decode): ");
        string mode = Console.ReadLine().ToLower();
        Console.Write("Enter text/code: ");
        string input = Console.ReadLine();

        if (mode == "encode")
            ChronoPulseEncoder.Main();
        else if (mode == "decode")
            ChronoPulseDecoder.Main();
        else
            Console.WriteLine("Invalid mode");
    }
}
